src/backend/cuda/cudnnModule.cpp

```diff
@@ -22,6 +22,22 @@ using std::string;
 
 namespace cuda {
 
+// Latest version from each minor releases are enlisted below
+static const std::vector<common::Version> cudnnVersions = {
+    // clang-format off
+    make_tuple(7, 6, 5),
+    make_tuple(7, 5, 1),
+    make_tuple(7, 4, 2),
+    make_tuple(7, 3, 1),
+    make_tuple(7, 2, 1),
+    make_tuple(7, 1, 4),
+    make_tuple(7, 0, 5),
+    make_tuple(6, 0, 21),
+    make_tuple(5, 1, 10),
+    make_tuple(4, 0, 7),
+    // clang-format off
+};
```

Comment:
would `constexpr std::array` work here ?

Reply:
I don't think so because DependencyModule needs to work for any library that we want to load and the number of versions we can enlist for any such library is not fixed. So, the DependencyModule constructor should take a vector. Given this, if we store the version list of the libraries in a std::array then, we need to convert that into a std::vector which will essentially result in a copy. This much worse than passing a reference to `static const` vector.

There is however another way if we had something like with `array_view` we which don't until C++20's `std::span`. Having something like this, we can accept this `array_view` via DependencyModule and it doesn't matter what container is used for version enlisting.

Reply:
We can use boost if there is something like `array_view` in there.

Reply:
It has boost::span